### PR TITLE
STYLE: Use the `itkNameOfTestExecutableMacro` macro in tests

### PR DESCRIPTION
**Pull request by jhlegarreta**
Use the `itkNameOfTestExecutableMacro` macro to get the test name
instead of hard-coding the test name.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Left behind in PR #2939.

A couple of additional notes:
- Not sure if this is the right move for `itkCMakeConfigurationTest`, as it does not depend on ITK. Insights are welcome.
- Some of the changes (added the `Missing parameters` message, changing `std::cout` to `std::cerr`) were only made in cases where the whole check was changed in order to restrict the number of changes to what it made sense as the necessary minimum. Making the style of the checking consistent beyond these can be done in a separate PR.

File: Modules/Core/Common/test/itkCMakeConfigurationTest.cxx [hunk 1]
```diff
@@ -74,7 +75,7 @@ main(int argc, char * argv[])
 {
   if (argc != 2)
   {
-    std::cerr << "Usage: itkCMakeInformationTest <top-of-build-tree>\n";
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " <top-of-build-tree>\n";
```
Comment (dzenanz) [inline Modules/Core/Common/test/itkCMakeConfigurationTest.cxx:78]: If this does not compile, renaming into `itkCMakeConfigurationTest` should be done.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/test/itkCMakeConfigurationTest.cxx (hunk 1 above)
Comment (dzenanz) [reply Modules/Core/Common/test/itkCMakeConfigurationTest.cxx:78]: Oh, it is used here:
https://github.com/InsightSoftwareConsortium/ITK/blob/24c3ad91db624592318c886c8d81614e0662ea72/Modules/Core/Common/test/CMakeLists.txt#L557-L560
and it did not get broken. All OK!

**Review by thewtex — APPROVED** [review]
:green_heart:

**Pull request merged**